diff -urb libpcap_notc/Win32/Prj/libpcap.dsp libpcap/Win32/Prj/libpcap.dsp
--- libpcap_notc/Win32/Prj/libpcap.dsp	2008-05-16 10:06:05.828125000 -0700
+++ libpcap/Win32/Prj/libpcap.dsp	2008-05-16 10:08:56.906250000 -0700
@@ -200,6 +200,14 @@
 # End Source File
 # Begin Source File
 
+SOURCE="..\..\pcap-tc.c"
+# End Source File
+# Begin Source File
+
+SOURCE="..\..\pcap-tc.h"
+# End Source File
+# Begin Source File
+
 SOURCE="..\..\Pcap-win32.c"
 # End Source File
 # Begin Source File
diff -urb libpcap_notc/fad-win32.c libpcap/fad-win32.c
--- libpcap_notc/fad-win32.c	2007-09-25 13:34:36.000000000 -0700
+++ libpcap/fad-win32.c	2008-05-16 10:08:56.828125000 -0700
@@ -43,6 +43,7 @@
 #include <pcap.h>
 #include <pcap-int.h>
 #include <Packet32.h>
+#include "pcap-tc.h"
 
 #include <errno.h>
 	
diff -urb libpcap_notc/gencode.c libpcap/gencode.c
--- libpcap_notc/gencode.c	2008-05-16 10:06:05.703125000 -0700
+++ libpcap/gencode.c	2008-05-16 10:05:03.468750000 -0700
@@ -374,6 +374,18 @@
 	const char * volatile xbuf = buf;
 	int len;
 
+	/*
+	 * We cannot generate PPI filters with TurboCap ports, as the
+	 * compiler doesn't generate the correct filtering code in case
+	 * of ethernet over PPI. It only generates 802.11 over PPI code
+	 */
+	if (p->TcInstance != NULL && p->linktype == DLT_PPI)
+	{
+		snprintf(p->errbuf, PCAP_ERRBUF_SIZE, "pcap_compile cannot generate filters for a TurboCap port when the PPI linktype is used.");
+		return -1;
+	
+	}
+
 #ifdef HAVE_REMOTE
 	/*
 	   Check if:
diff -urb libpcap_notc/pcap-int.h libpcap/pcap-int.h
--- libpcap_notc/pcap-int.h	2008-05-16 10:06:05.703125000 -0700
+++ libpcap/pcap-int.h	2008-05-16 10:08:56.843750000 -0700
@@ -48,6 +48,7 @@
 
 #ifdef WIN32
 #include <Packet32.h>
+#include "pcap-tc.h"
 #endif /* WIN32 */
 
 #ifdef MSDOS
@@ -231,6 +232,12 @@
 	ADAPTER *adapter;
 	LPPACKET Packet;
 	int nonblock;
+	/* TurboCap specific information */
+	TC_INSTANCE TcInstance;
+	TC_PACKETS_BUFFER TcPacketsBuffer;
+	ULONG	TcAcceptedCount;
+	PCHAR   PpiPacket;
+
 #else
 	int fd;
 	int selectable_fd;
diff -urb libpcap_notc/pcap-win32.c libpcap/pcap-win32.c
--- libpcap_notc/pcap-win32.c	2008-05-16 10:06:05.718750000 -0700
+++ libpcap/pcap-win32.c	2008-05-16 10:08:56.843750000 -0700
@@ -56,6 +56,8 @@
 #include <pcap-remote.h>
 #endif	/* HAVE_REMOTE */
 
+#include "pcap-tc.h"
+
 static int pcap_setfilter_win32_npf(pcap_t *, struct bpf_program *);
 static int pcap_setfilter_win32_dag(pcap_t *, struct bpf_program *);
 static int pcap_getnonblock_win32(pcap_t *, char *);
@@ -812,7 +814,15 @@
 	if (p == NULL)
 		return (NULL);
 
+	if (IsTcDevice(p) == TRUE)
+	{
+		p->activate_op = TcActivate;
+	}
+	else
+	{
 	p->activate_op = pcap_activate_win32;
+	}
+
 	return (p);
 }
 
@@ -906,5 +916,5 @@
 int
 pcap_platform_finddevs(pcap_if_t **alldevsp, char *errbuf)
 {
-	return (0);
+	return TcFindAllDevs(alldevsp, errbuf);
 }
