diff -urb libpcap/Win32/Prj/libpcap.dsp libpcap_tc/Win32/Prj/libpcap.dsp
--- libpcap/Win32/Prj/libpcap.dsp	2008-04-30 15:44:39.406250000 -0700
+++ libpcap_tc/Win32/Prj/libpcap.dsp	2008-04-30 15:41:49.093750000 -0700
@@ -200,6 +200,14 @@
 # End Source File
 # Begin Source File
 
+SOURCE="..\..\pcap-tc.c"
+# End Source File
+# Begin Source File
+
+SOURCE="..\..\pcap-tc.h"
+# End Source File
+# Begin Source File
+
 SOURCE="..\..\Pcap-win32.c"
 # End Source File
 # Begin Source File
diff -urb libpcap/fad-win32.c libpcap_tc/fad-win32.c
--- libpcap/fad-win32.c	2007-09-25 13:34:36.000000000 -0700
+++ libpcap_tc/fad-win32.c	2008-04-25 14:13:35.984375000 -0700
@@ -43,6 +43,7 @@
 #include <pcap.h>
 #include <pcap-int.h>
 #include <Packet32.h>
+#include "pcap-tc.h"
 
 #include <errno.h>
 	
diff -urb libpcap/pcap-int.h libpcap_tc/pcap-int.h
--- libpcap/pcap-int.h	2008-04-30 15:44:39.234375000 -0700
+++ libpcap_tc/pcap-int.h	2008-04-30 11:43:28.531250000 -0700
@@ -48,6 +48,7 @@
 
 #ifdef WIN32
 #include <Packet32.h>
+#include "pcap-tc.h"
 #endif /* WIN32 */
 
 #ifdef MSDOS
@@ -231,6 +232,12 @@
 	ADAPTER *adapter;
 	LPPACKET Packet;
 	int nonblock;
+	/* TurboCap specific information */
+	TC_INSTANCE TcInstance;
+	TC_PACKETS_BUFFER TcPacketsBuffer;
+	ULONG	TcAcceptedCount;
+	PCHAR   PpiPacket;
+
 #else
 	int fd;
 	int selectable_fd;
diff -urb libpcap/pcap-win32.c libpcap_tc/pcap-win32.c
--- libpcap/pcap-win32.c	2008-04-30 15:44:39.250000000 -0700
+++ libpcap_tc/pcap-win32.c	2008-04-30 11:43:28.546875000 -0700
@@ -56,6 +56,8 @@
 #include <pcap-remote.h>
 #endif	/* HAVE_REMOTE */
 
+#include "pcap-tc.h"
+
 static int pcap_setfilter_win32_npf(pcap_t *, struct bpf_program *);
 static int pcap_setfilter_win32_dag(pcap_t *, struct bpf_program *);
 static int pcap_getnonblock_win32(pcap_t *, char *);
@@ -784,7 +785,15 @@
 	if (p == NULL)
 		return (NULL);
 
+	if (IsTcDevice(p) == TRUE)
+	{
+		p->activate_op = TcActivate;
+	}
+	else
+	{
 	p->activate_op = pcap_activate_win32;
+	}
+
 	return (p);
 }
 
@@ -878,5 +887,5 @@
 int
 pcap_platform_finddevs(pcap_if_t **alldevsp, char *errbuf)
 {
-	return (0);
+	return TcFindAllDevs(alldevsp, errbuf);
 }
